### DISCUSSION TITLE
第2节课作业(19班)

### DISCUSSION THREAD

**Discussion by LYMDLUT** [InternLM/Tutorial#105] [作业]
第2节课作业(19班)

**Comment by Sixdes** [comment]
https://github.com/Sixdes/puyu-course/tree/main/homework0108

**Comment by hu-qi** [comment]
## 基础作业：

-  使用 InternLM-Chat-7B 模型生成 300 字的小故事（需截图）。

```请创作一个300字的小故事，故事的主题是人工智能战胜人类```

![image](https://github.com/InternLM/tutorial/assets/17986122/19d40234-f121-429b-a7cd-13014d3b3754)




-  熟悉 hugging face 下载功能，使用 huggingface_hub python 包，下载 InternLM-20B 的 config.json 文件到本地（需截图下载过程）。

网络问题无法下载？
![image](https://github.com/InternLM/tutorial/assets/17986122/799145b2-1567-4b2b-a8f7-75779ee5f887)

通过Q&A的提示，修改镜像得以下载成功：

```
export HF_ENDPOINT=https://hf-mirror.com  && python down_model.py 
```

![image](https://github.com/InternLM/tutorial/assets/17986122/aa0c3f27-2d4f-41af-8ca6-9206557325a1)




## 进阶作业（可选做）

- 完成浦语·灵笔的图文理解及创作部署（需截图）

![image](https://github.com/InternLM/tutorial/assets/17986122/0b4574da-2fe2-4534-9032-0f2413aefc9e)



![image](https://github.com/InternLM/tutorial/assets/17986122/ade167cb-d7c2-48cb-b170-f67067aee728)



-  完成 Lagent 工具调用 Demo 创作部署（需截图）

```
鸡兔同笼问题，假设鸡兔同笼，上有35个头，下有94只足，问鸡兔分别有几只？
```

![image](https://github.com/InternLM/tutorial/assets/17986122/b11ee8d9-1a65-4cc3-8563-f3f90d00c240)

**Comment by Jiangzzzzl** [comment]
<img width="1267" alt="1c44b0c078b1383bb4416e20f344943" src="https://github.com/InternLM/tutorial/assets/150671792/97fa4523-363f-4407-ba50-e866b295ca25">
<img width="1244" alt="d5107bd480ae3830ba3caee5fec6092" src="https://github.com/InternLM/tutorial/assets/150671792/004f69e6-6b37-4792-8575-77cf82fc96a7">
<img width="1274" alt="f789ecb3846c0a9b2acd461b2b58d0a" src="https://github.com/InternLM/tutorial/assets/150671792/7ad14ebb-67bb-4126-9c1c-9941c3f61189">
<img width="1274" alt="caa08fa4e1792274287f950046612d4" src="https://github.com/InternLM/tutorial/assets/150671792/677149df-166a-4208-9bfd-425443fab0aa">
<img width="1267" alt="03fc12b9f92a84109fb26471d50d1a6" src="https://github.com/InternLM/tutorial/assets/150671792/d48f958a-4113-442d-bdfd-5f80f439cb8f">

**Comment by jts250** [comment]
https://github.com/jts250/homework_llm/blob/main/homework1.pdf

**Comment by JankinTian** [comment]
https://blog.csdn.net/xiao_xian_/article/details/135442608

**Comment by romanwin28** [comment]
[Uploading internlm_2.pdf…]()

**Reply by romanwin28** [reply]
当时着急没上传好，补上传：
[internlm_2.pdf](https://github.com/InternLM/tutorial/files/14071002/internlm_2.pdf)

**Comment by WiselyC0430** [comment]
![130950b08c4d82e96de126ce1f9bd49](https://github.com/InternLM/tutorial/assets/96457986/5ebd82c8-eca7-4492-a3e5-ea72986a7081)
![b8dd3d6947332da09486efcf75c91fa](https://github.com/InternLM/tutorial/assets/96457986/f9126384-f08b-4b2d-bd65-f613371a2131)
出现了复读机的问题。

**Comment by jia-heng** [comment]
https://github.com/jia-heng/puyu-learn/blob/main/lesson2.md

**Comment by AllYoung** [comment]
https://blog.csdn.net/sunshine_youngforyou/article/details/135440349

**Comment by melonwine** [comment]
https://github.com/melonwine/InternLMPlay/blob/master/02/asgmt02.md

**Comment by SnowKingLiu** [comment]
https://github.com/SnowKingLiu/puyu/blob/main/course2/Course2.md

**Comment by mkdirmushroom** [comment]
[第2节课作业](https://github.com/mkdirmushroom/note/blob/main/InternLM/tutorial1_assignment.md)

**Comment by Hamasuta666** [comment]
第2节课笔记（作业在文档末尾）：
https://zhuanlan.zhihu.com/p/680561771
